### PR TITLE
Embed api

### PR DESCRIPTION
**Pull request by joshkurz**
* adds a new embed api
* decouples graph logic from embed api logic

### CODE REVIEW

**Comment by brettlangdon** [comment]
Test *should* fail, this file will need to be updated (renamed to `embed.js` and imports/names fixed): https://github.com/brettlangdon/node-dogapi/blob/e58e06bfa8f4fde1262e87cb4970da5600f28468/test/api/graph.js

**Comment by brettlangdon** [comment]
other than that, quick glance it looks decent, I won't be able to give it an in-depth look until later tonight or tomorrow morning

**Comment by brettlangdon** [comment]
lastly... thanks again for your contribution!!!!!! :)

File: lib/api/embed.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    create: create,
+    revoke: revoke,
+    getAll: getAll,
+    get: get,
+    getUsage: function(){
+        return [
+            "  dogapi embed create <embed_json> [--timeframe <timeframe>] [--size <size>] [--legend <legend>] [--title <title>]",
+            "  dogapi embed revoke <embed_id>",
+            "  dogapi embed get <emebd_id>",
+            "  dogapi embed getall <emebd_id>"
```
Comment (brettlangdon) [inline lib/api/embed.js]: shouldn't need `<embed_id>` here.

File: lib/api/embed.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+module.exports = {
+    create: create,
+    revoke: revoke,
+    getAll: getAll,
+    get: get,
+    getUsage: function(){
+        return [
+            "  dogapi embed create <embed_json> [--timeframe <timeframe>] [--size <size>] [--legend <legend>] [--title <title>]",
+            "  dogapi embed revoke <embed_id>",
+            "  dogapi embed get <emebd_id>",
```
Comment (brettlangdon) [inline lib/api/embed.js]: `s/emebd/embed/`

File: lib/api/embed.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            "  dogapi embed getall <emebd_id>"
+        ];
+    },
+    getHelp: function(){
+        return [
+            "Embed:",
+            "  Subcommands:",
+            "    create <emebd_json> --timeframe <timeframe> --size <size> --legend <legend> --title <title>     |     create a new graph embed",
```
Comment (brettlangdon) [inline lib/api/embed.js]: `s/emebd/embed/`

File: lib/api/graph.js [hunk 4]
```diff
@@ -123,13 +56,11 @@ module.exports = {
             "  Subcommands:",
             "    snapshot <query> <from> <to> --events <event-query>     |     take a snapshot of a graph",
             "    create_embed <graph_json> --timeframe <timeframe> --size <size> --legend <legend> --title <title>     |     create a new graph embed",
+            "    revoke_embed <embed_id>     |     revoke/delete an embed",
+            "    get <embed_id>              |     gets a single embed object",
+            "    getall                      |     gets all embed objects",
```
Comment (brettlangdon) [inline lib/api/graph.js]: these three lines are no longer needed.

**Comment by brettlangdon** [comment]
@joshkurz sorry it took me so long to get around to it, but after my comments, this PR should be good to go.

**Comment by joshkurz** [comment]
updated and rebased

**Comment by brettlangdon** [comment]
I had one final thought here, can we link `dogapi.graph.createEmbed` to `dogapi.embed.create`, this way we don't have to bump major versions for this release. 

e.g.

```js
// graph.js
var embed = require('../embed');

module.exports = {
    createEmbed: embed.create,
    ...
};
```


Thoughts?

**Comment by joshkurz** [comment]
yeah that's a good idea.

**Comment by joshkurz** [comment]
I added the graph.js tests back also.

File: test/api/graph.js [hunk 5]
```diff
@@ -1,3 +1,19 @@
+Skip to content
```
Comment (brettlangdon) [inline test/api/graph.js]: Hmm, I think something isn't right here.

Also, we probably don't need to have the graph tests for `createEmbed`, we are reliably testing that method in another place.

**Comment by brettlangdon** [comment]
If you drop the graph createEmbed tests, then lgtm

**Comment by joshkurz** [comment]
dont you want to make sure its backwards compatible?

**Comment by brettlangdon** [comment]
Sure, but we shouldn't need to have duplicate tests to ensure that. We could add a test to assert that `dogapi.graph.createEmbed` exists, but probably not worth having a test just for that.

**Comment by joshkurz** [comment]
ok well they passed, I honestly just wanted to test to make sure it works at least once to be sure.

**Comment by brettlangdon** [comment]
:+1: great, one these latest tests pass, I'll merge and release right away!

Thanks your your contribution! I really appreciate the work!

**Pull request merged**